src/v/compat/metadata_dissemination_compat.h

```diff
@@ -88,4 +88,31 @@ struct compat_check<cluster::update_leadership_request_v2> {
     }
 };
 
+/*
+ * cluster::update_leadership_reply
+ */
+template<>
+struct compat_check<cluster::update_leadership_reply> {
+    static constexpr std::string_view name = "cluster::update_leadership_reply";
+    static std::vector<cluster::update_leadership_reply> create_test_cases() {
+        return generate_instances<cluster::update_leadership_reply>();
+    }
+    static void to_json(
+      [[maybe_unused]] cluster::update_leadership_reply obj,
```

Comment:
instead of `[[maybe_unused]]` you can just remove the variable names to satisfy the warnings, since they are definitely unused